src/core/MidiIO.cpp

```diff
@@ -50,7 +50,7 @@ void MidiIO::baseFromJson(json_t *rootJ) {
 
 std::vector<std::string> MidiIO::getDevices() {
 	/* Note: we could also use an existing interface if one exists */
-	static RtMidiIn *m = new RtMidiIn();
```

Comment:
`m` is never deleted, so it leaks one everytime `getDevices()` is called.
It leaked when it was static too, but only one copy.
Could either delete `m` at the end of the function, or just use `RtMidiIn m;`

Reply:
Or rather
```cpp
static RtMidiIn *m = NULL;
if (!m)
	m = new RtMidiIn();
```
to be explicit.

Reply:
Still leaks, but only one copy over the life of the program. Not a big deal but can be annoying if you start using memory analysis tools

Reply:
Yeah both work but I'd go with the `RtMidiIn m` solution. It looks clean and the function is not called very often so this is not relevant for performance.

Reply:
I added a proper check to see whether there already is a device and use that.